src/org/omegat/util/net/HttpConnectionUtils.java

```diff
@@ -0,0 +1,398 @@
+ **************************************************************************/
+
+package org.omegat.util.net;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.omegat.util.Log;
+import org.omegat.util.Preferences;
+import org.omegat.util.StaticUtils;
+import org.omegat.util.StringUtil;
+
+
+/**
+ * Utility collection for http connections.
+ *
+ * @author Kim Bruning
+ * @author Alex Buloichik (alex73mail@gmail.com)
+ * @author Didier Briel
+ * @author Rashid Umarov
+ * @author Hiroshi Miura
+ */
+public final class HttpConnectionUtils {
+    protected static final String DEFAULT_RESPONSE_CHARSET = "ISO-8859-1";
+    protected static final String CHARSET_MARK = "charset=";
+    /**
+     * Buffer size for downloading.
+     */
+    private static final int BUFFER_SIZE = 8196;
+
+    /**
+     * default timeout
+     */
+    private static final int TIMEOUT_MS = 10_000;
+
+    /**
+     * Don't instanciate util class.
+     */
+    private HttpConnectionUtils() {}
+
+    /**
+     * Download a file to memory
+     */
+    public static String getURL(URL url) throws IOException {
+        URLConnection urlConn = url.openConnection();
+        urlConn.setConnectTimeout(TIMEOUT_MS);
+        urlConn.setReadTimeout(TIMEOUT_MS);
+        try (InputStream in = urlConn.getInputStream()) {
+            return IOUtils.toString(in, StandardCharsets.UTF_8);
+        }
+    }
+
+    public static void downloadZipFileAndExtract(URL url, File dir, List<String> expectedFiles) throws IOException {
+        URLConnection conn = url.openConnection();
+        conn.setConnectTimeout(TIMEOUT_MS);
+        conn.setReadTimeout(TIMEOUT_MS);
+        try (InputStream in = conn.getInputStream()) {
+            List<String> extracted = StaticUtils.extractFromZip(in, dir, expectedFiles::contains);
+            if (!expectedFiles.containsAll(extracted)) {
+                throw new FileNotFoundException("Could not extract expected files from zip; expected: "
+                        + expectedFiles + ", extracted: " + extracted);
+            }
+        } catch (IllegalArgumentException ex) {
+            throw new FlakyDownloadException(ex);
+        }
+    }
+
+    /**
+     * Downloads a binary file from a URL
+     * @param jarFileURL HTTP URL of the file to be downloaded
+     * @param saveFilePath path of the file
+     * @throws IOException
+     */
+    public static boolean downloadBinaryFile(final URL jarFileURL, final Map<String, String> headers,
```

Comment:
Where did this method come from? It isn't used anywhere.

Is this something you need for #69?

Reply:
> Is this something you need for #69?

Yes, I plan to use it to download Jar file from repository.